src/main/java/org/opensearch/plugin/insights/core/service/store/PriorityQueueTopQueriesStore.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.plugin.insights.core.service.store;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.PriorityBlockingQueue;
+
+/**
+ * Implementation of {@link TopQueriesStore} that uses a {@link PriorityBlockingQueue}
+ * to manage Top N search query records or groups. This implementation is thread-safe.
+ *
+ * @param <T> The type of records that this store will manage.
+ */
+public class PriorityQueueTopQueriesStore<T> implements TopQueriesStore<T> {
```

Comment:
This implementation and the corresponding interface is not helping IMO. The interface is tightly coupled with the Priority Queue implementation and even this implementation is very thin wrapper on top of Priority Queue. Maybe we can remove this for now, think more carefully the interface and default implementation

Reply:
+1. TopQueriesStore exposes methods similar to a queue so it doesn't make much sense to have this. My idea was to have a topQueriesStore designed in a way where you can use it to store topN queries on disk/remote etc(or different sinks) and can extend/plug it easily in future if needed. So better to remove this altogether for now.

Reply:
Removed the interface.